cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSISQLUtils.java

```diff
@@ -0,0 +1,268 @@
+                timestampFormat,
+                sqlInstance,
+                destination).toString();
+
+        PreparedStatement previousStatement = connection.prepareStatement(query);
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = addJsonValues(previousStatement,
+                    lastData,
+                    attrNativeTypes);
+        } catch (SQLException e) {
+            LOGGER.error(sqlInstance + " SQLEXCEPTION Error creating upsert statement " + e);
+        } catch (Exception e) {
+            LOGGER.error(sqlInstance + " GENERICEXCEPTION Error creating upsert statement " + e);
+        }
+        LOGGER.info("[NGSISQLUtils.upsertStatement] PreparedStatement for upsert created successfully, all batches added. " + query);
+        return preparedStatement;
+
+    }
+
+    /**
+     * Sql upsert query string buffer.
+     *
+     * @param aggregation     the aggregation
+     * @param lastData        the last data
+     * @param tableName       the table name
+     * @param tableSuffix     the table suffix
+     * @param uniqueKey       the unique key
+     * @param timestampKey    the timestamp key
+     * @param timestampFormat the timestamp format
+     * @param sqlInstance     the sql instance
+     * @param destination     the destination
+     * @return the string buffer
+     */
+    protected static StringBuffer sqlUpsertQuery(LinkedHashMap<String, ArrayList<JsonElement>> aggregation,
+                                              LinkedHashMap<String, ArrayList<JsonElement>> lastData,
+                                              String tableName,
+                                              String tableSuffix,
+                                              String uniqueKey,
+                                              String timestampKey,
+                                              String timestampFormat,
+                                              String sqlInstance,
+                                              String destination) {
+
+        StringBuffer fieldsForInsert;
+        StringBuffer valuesForInsert = sqlQuestionValues(lastData.keySet());
+        StringBuffer updateSet = new StringBuffer();
+        StringBuffer postgisTempReference = new StringBuffer("EXCLUDED");
+        StringBuffer postgisDestination = new StringBuffer(destination).append(".").append(tableName).append(tableSuffix);
+        StringBuffer query = new StringBuffer();
+        boolean first = true;
+
+        for (String key : lastData.keySet()) {
+            if (!key.equals(uniqueKey) && first) {
+                updateSet.append(key).append("=").append(postgisTempReference).append(".").append(key);
+                first = false;
+            } else if (!key.equals(uniqueKey)) {
+                updateSet.append(", ").append(key).append("=").append(postgisTempReference).append(".").append(key);
+            }
+        }
+
+        if (sqlInstance.equals("postgresql")) {
```

Comment:
That's correct. In both cases this String `sqlInstance` is provided by the sink. In Postgis and PostgreSQL is handled as `postgresql `. This is because the driver used to connect to the database is the same one.